src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -363,10 +385,13 @@ private function createFirewall(ContainerBuilder $container, $id, $firewall, &$a
         // Authentication listeners
         list($authListeners, $defaultEntryPoint) = $this->createAuthenticationListeners($container, $id, $firewall, $authenticationProviders, $defaultProvider, $configuredEntryPoint);
 
+        $config->replaceArgument(6, $configuredEntryPoint ?: $defaultEntryPoint);
```

Comment:
Maybe `anonymous` could be a boolean parameter of the config to be exposed in the collector.